lib/Dialect/AIE/IR/AIEDialect.cpp

```diff
@@ -1545,60 +1547,93 @@ LogicalResult DMABDOp::verify() {
   if (!isa<BufferOp, ExternalBufferOp>(getBuffer().getDefiningOp()))
     return emitOpError(
         "BDs only support BufferOp or ExternalBufferOp operands.");
-  if (auto memOp = getOperation()->getParentOfType<MemOp>()) {
+
+  if (getLenInBytes() % 4)
+    return emitOpError("transfer length must be multiple of 4B (32b)");
+
+  if (getOperation()->getParentOfType<MemOp>()) {
+    auto memOp = getOperation()->getParentOfType<MemOp>();
     if (auto bufferOp = getBufferOp();
         bufferOp.getTileOp().colIndex() != memOp.colIndex() ||
         bufferOp.getTileOp().rowIndex() != memOp.rowIndex())
-      return emitOpError("can only access a buffer in the same tile.");
+      return emitOpError(
+          "Core tile DMAs can only access a buffer in the same tile.");
+    if (std::optional<int32_t> bdId = getBdId(); bdId.has_value() && *bdId > 15)
```

Comment:
Factor out magic constant 15, preferably to the TargetModel

Reply:
there's actually a verifyTraint/interface thing that does this check against the target model but dma_bd doesn't implement. Let me try to implement that interface instead of this.